### PR TITLE
SAM-2669 - Add Samigo events constants to api to avoid duplication

### PR DESCRIPTION
**Pull request by jonespm**

### CODE REVIEW

**Comment by jonespm** [comment]
Wanted to see how all the events in Samigo looked before looking at the change of the event names in SAM-3012. Can't believe how many events there ended up being.

File: samigo/samigo-api/src/java/org/sakaiproject/samigo/util/SamigoConstants.java [hunk 1]
```diff
@@ -40,14 +40,60 @@
     /*
      * Events
      */
+    // Submission events
```
Comment (ern) [inline samigo/samigo-api/src/java/org/sakaiproject/samigo/util/SamigoConstants.java]: While your here should probably make the class final and add a private constructor so this class cannot be instantiated! This is usually the best accepted pattern for classes containing constants.

Comment (buckett) [inline samigo/samigo-api/src/java/org/sakaiproject/samigo/util/SamigoConstants.java]: The alternative is to make the class an interface instead.

File: samigo/samigo-api/src/java/org/sakaiproject/samigo/util/SamigoConstants.java [hunk 2]
```diff
@@ -40,14 +40,60 @@
...
+    public static final     String      EVENT_ASSESSMENT_REMOVE                             = "sam.assessment.remove";
+    public static final     String      EVENT_ASSESSMENT_REVISE                             = "sam.assessment.revise";
+    public static final     String      EVENT_ASSESSMENT_UNINDEXITEM                        = "sam.assessment.unindexitem";
+    public static final     String      EVENT_ASSESSMENT_PUBLISH                            = "sam.assessment.publish";
+
+    //Published assessment events
+    public static final     String      EVENT_PUBLISHED_ASSESSMENT_REVISE                   = "sam.pubassessment.revise";
+    public static final     String      EVENT_PUBLISHED_ASSESSMENT_SAVEITEM                 = "sam.pubassessment.saveitem";
+    public static final     String      EVENT_PUBLISHED_ASSESSMENT_REMOVE                   = "sam.pubAssessment.remove";
+    public static final     String      EVENT_PUBLISHED_ASSESSMENT_UNPUBLISH                = "sam.pubassessment.confirm_edit";
```
Comment (buckett) [inline samigo/samigo-api/src/java/org/sakaiproject/samigo/util/SamigoConstants.java]: Is `UNPUBLISHED` a good constant for `.confirm_edit`?

Comment (jonespm) [inline samigo/samigo-api/src/java/org/sakaiproject/samigo/util/SamigoConstants.java]: Thanks, I'll double check that. I looked at the commit 3-4 times to make sure that the events seemed accurate but feel like one of these might have been missed. Hope it was just my bad naming of a variable. ;)

File: samigo/samigo-api/src/java/org/sakaiproject/samigo/util/SamigoConstants.java (hunk 1 above)
Comment (jonespm) [inline samigo/samigo-api/src/java/org/sakaiproject/samigo/util/SamigoConstants.java]: Yeah, I can update that in here.

**Comment by jonespm** [comment]
Fixed the two issues mentioned. @ern suggestion of final was preferred over interface on StackOverflow just to prevent some programmer from actually trying to implement it

File: samigo/samigo-api/src/java/org/sakaiproject/samigo/util/SamigoConstants.java (hunk 2 above)
Comment (buckett) [inline samigo/samigo-api/src/java/org/sakaiproject/samigo/util/SamigoConstants.java]: I often get code blindness when attempting to read changes like this, it's really hard to keep your mind focused.

**Pull request merged**